Not working on MC 1.5.2 with Buildcraft 3.5.2 or 3.5.3 (API change?)

Minecraft 1.5.2 fails to start with the latest versions of Buildcraft (3.5.3) and LogisticsPipes (0.7.2.dev.423) on Minecraft 1.5.2. Based on the crash report, it appears to be an API change. This was tested with the latest Forge and no other mods installed. Here's the crash report:

---- Minecraft Crash Report ----
// I'm sorry, Dave.

Time: 5/15/13 3:28 PM
Description: Failed to start game

cpw.mods.fml.common.LoaderException: java.lang.NoSuchFieldError: pipes
    at cpw.mods.fml.common.LoadController.transition(LoadController.java:142)
    at cpw.mods.fml.common.Loader.initializeMods(Loader.java:696)
    at cpw.mods.fml.client.FMLClientHandler.finishMinecraftLoading(FMLClientHandler.java:206)
    at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:448)
    at net.minecraft.client.MinecraftAppletImpl.func_71384_a(SourceFile:56)
    at net.minecraft.client.Minecraft.run(Minecraft.java:733)
    at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NoSuchFieldError: pipes
    at logisticspipes.proxy.buildcraft.BuildCraftProxy.registerPipe(BuildCraftProxy.java:257)
    at logisticspipes.proxy.buildcraft.BuildCraftProxy.createPipe(BuildCraftProxy.java:269)
    at logisticspipes.proxy.buildcraft.BuildCraftProxy.registerPipes(BuildCraftProxy.java:211)
    at logisticspipes.LogisticsPipes.PostLoad(LogisticsPipes.java:354)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at cpw.mods.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:494)
    at sun.reflect.GeneratedMethodAccessor3.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at com.google.common.eventbus.EventHandler.handleEvent(EventHandler.java:74)
    at com.google.common.eventbus.SynchronizedEventHandler.handleEvent(SynchronizedEventHandler.java:45)
    at com.google.common.eventbus.EventBus.dispatch(EventBus.java:314)
    at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:296)
    at com.google.common.eventbus.EventBus.post(EventBus.java:267)
    at cpw.mods.fml.common.LoadController.propogateStateMessage(LoadController.java:165)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
    at java.lang.reflect.Method.invoke(Unknown Source)
    at com.google.common.eventbus.EventHandler.handleEvent(EventHandler.java:74)
    at com.google.common.eventbus.SynchronizedEventHandler.handleEvent(SynchronizedEventHandler.java:45)
    at com.google.common.eventbus.EventBus.dispatch(EventBus.java:314)
    at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:296)
    at com.google.common.eventbus.EventBus.post(EventBus.java:267)
    at cpw.mods.fml.common.LoadController.distributeStateMessage(LoadController.java:98)
    at cpw.mods.fml.common.Loader.initializeMods(Loader.java:695)
    ... 5 more

A detailed walkthrough of the error, its code path and all known details is as follows:
(dashes removed to prevent Markdown parsing)

-- System Details --
Details:
    Minecraft Version: 1.5.2
    Operating System: Windows 8 (amd64) version 6.2
    Java Version: 1.7.0_17, Oracle Corporation
    Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
    Memory: 554816304 bytes (529 MB) / 669253632 bytes (638 MB) up to 954466304 bytes (910 MB)
    JVM Flags: 2 total; -Xms512m -Xmx1024m
    AABB Pool Size: 0 (0 bytes; 0 MB) allocated, 0 (0 bytes; 0 MB) used
    Suspicious classes: FML and Forge are installed
    IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
    FML: MCP v7.51 FML v5.2.6.700 Minecraft Forge 7.8.0.700 10 mods loaded, 10 mods active
    mcp{7.44} [Minecraft Coder Pack](minecraft.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized
    FML{5.2.6.700} [Forge Mod Loader](coremods) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized
    Forge{7.8.0.700} [Minecraft Forge](coremods) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized
    BuildCraft|Core{3.5.3} [BuildCraft](buildcraft-A-3.5.3.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized
    BuildCraft|Builders{3.5.3} [BC Builders](buildcraft-A-3.5.3.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized
    BuildCraft|Energy{3.5.3} [BC Energy](buildcraft-A-3.5.3.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized
    BuildCraft|Factory{3.5.3} [BC Factory](buildcraft-A-3.5.3.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized
    BuildCraft|Transport{3.5.3} [BC Transport](buildcraft-A-3.5.3.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized
    BuildCraft|Silicon{3.5.3} [BC Silicon](buildcraft-A-3.5.3.jar) Unloaded->Constructed->Pre-initialized->Initialized->Post-initialized
    LogisticsPipes|Main{0.7.2.dev.423} [Logistics Pipes](LogisticsPipes-MC1.5.2-0.7.2.dev.423.jar) Unloaded->Constructed->Pre-initialized->Initialized->Errored
    LWJGL: 2.4.2
    OpenGL: NVS 4200M/PCIe/SSE2 GL version 4.3.0, NVIDIA Corporation
    Is Modded: Definitely; Client brand changed to 'fml,forge'
    Type: Client (map_client.txt)
    Texture Pack: Default
    Profiler Position: N/A (disabled)
    Vec3 Pool Size: ~~ERROR~~ NullPointerException: null

--- a/common/logisticspipes/proxy/buildcraft/BuildCraftProxy.java
+++ b/common/logisticspipes/proxy/buildcraft/BuildCraftProxy.java
@@ -12,6 +12,7 @@ import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 
 import logisticspipes.LogisticsPipes;
 import logisticspipes.config.Configs;
@@ -254,7 +255,19 @@ public class BuildCraftProxy {
 	public static ItemPipe registerPipe(int key, Class<? extends Pipe> clas) {
 		ItemPipe item = new ItemLogisticsPipe(key, clas);
 
-		BlockGenericPipe.pipes.put(item.itemID, clas);
+		Map<Integer, Class<? extends Pipe>> pipes = null;
+		
+		try {
+			pipes = BlockGenericPipe.pipes;
+		} catch(NoSuchFieldError e) {
+			try {
+				pipes = (Map<Integer, Class<? extends Pipe>>) BlockGenericPipe.class.getDeclaredField("pipes").get(null);
+			} catch (Exception e2) {
+				return null;
+			}
+		}
+		
+		pipes.put(item.itemID, clas);
 
 		Pipe dummyPipe = BlockGenericPipe.createPipe(item.itemID);
 		if (dummyPipe != null) {

